streaming/src/main/scala/org/apache/gearpump/streaming/appmaster/TaskManager.scala

```diff
@@ -121,7 +121,7 @@ private[appmaster] class TaskManager(
 
     case Event(RegisterTask(taskId, executorId, host), state@ TaskRegistrationState(register)) =>
       val client = sender
-      getMinClock.map(client ! StartClock(_))
+      getLatestStartClock.map(client ! StartClock(_))
```

Comment:
I think we should replay from `LatestStartClock`. 
```scala
private def latestStartClock: TimeStamp = {
   minCheckpointTime.getOrElse(minClock)
}
```
when it's at least once, `LatestStartClock` returns the value of `LatestMinClock` so it covers both at least once and exact once for replay.